x/btcstaking/keeper/msg_server.go

```diff
@@ -401,6 +401,16 @@ func (ms msgServer) AddCovenantSigs(goCtx context.Context, req *types.MsgAddCove
 		return nil, types.ErrInvalidCovenantPK.Wrapf("covenant pk: %s", req.Pk.MarshalHex())
 	}
 
+	if btcDel.IsSignedByCovMember(req.Pk) || btcDel.BtcUndelegation.IsSignedByCovMember(req.Pk) {
```

Comment:
Should this be

```suggestion
	if btcDel.IsSignedByCovMember(req.Pk) && btcDel.BtcUndelegation.IsSignedByCovMember(req.Pk) {
```

which is the expected behaviour of a covenant member. If the covenant member only submits one of these signatures then the message will be rejected anyway

Reply:
tbh I do not see much difference between using either `&&` or `||`, adding both types of signatures should be atomic so in theory it would be enough to check only `btcDel.IsSignedByCovMember(req.Pk)` . 

This makes me think that there can be more refactoring done here. 

I will change it to`&&` for now as this is maybe more complete.